### PR TITLE
Change how batch textures are counted in `ModelExperimentalStatistics`

### LINKED ISSUE #10536: Refactor `ModelExperimentalStatistics`
In #10530 two unit tests in `Cesium3DTilesetSpec` broke after using `ModelExperimental` to load tilesets:

- "verify memory usage statistics"
- "verify memory usage statistics for shared resources"

The first one happens because `PrimitiveStatisticsPipelineStage` counts the bytelength of the feature tables before the `batchTextures` are done loading. The pipeline is never rerun after they load in, so they never get counted by the model. To refactor this, we'll want to make the `propertyTablesByteLength` getter iterate over the feature tables and count the statistics, so that they can accommodate the textures as they load in.

The second one has to do with counting shared resources, but we have not looked into it deeply yet.

### PR DESCRIPTION
**Pull request by j9liu**
Fixes #10536. Batch textures are now queried for their memory each frame, instead of adding a hardcoded value that will later be changed. This is to account for batch / pick textures being created dynamically.

@ptrgags I think some things got lost in translation in our offline conversation -- I wasn't sure if the ID system was necessary for batch textures on the `ModelExperimentalStatistics` level, let me know if not and I'll remove them.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @j9liu!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/refactor-model-experimental-statistics/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: Source/Scene/ModelExperimental/ModelExperimentalStatistics.js [hunk 1]
```diff
@@ -30,9 +31,9 @@ export default function ModelExperimentalStatistics() {
 
   /**
    * Total size of all geometry buffers in bytes. This accounts for the vertex
-   * attributes (which includes feature IDs and property attributes),
-   * and index buffers of all the model's primitives. Any attributes generated
-   * by the pipeline are included in this total.
+   * attributes (which includes feature IDs and property attributes) and index
+   * buffers of all the model's primitives. Any attributes generate by the
```
Comment (j9liu) [inline Source/Scene/ModelExperimental/ModelExperimentalStatistics.js:35]:
```suggestion
   * buffers of all the model's primitives. Any attributes generated by the
```

File: Specs/Scene/Cesium3DTilesetSpec.js [hunk 2]
```diff
@@ -1110,7 +1108,6 @@ describe(
       // * Two i3dm tiles with external glTF - shared resources
       // Expect to see some saving with memory usage since two of the tiles share resources
       // All tiles reference the same external texture but texture caching is not supported yet
-      // TODO : tweak test when #5051 is in
```
Comment (j9liu) [inline Specs/Scene/Cesium3DTilesetSpec.js:1113]: think I accidentally deleted this, I'll add it back in

**Comment by j9liu** [comment]
I'm looking into the other unit test, `"verify memory usage statistics for shared resources"`... we can't replicate this with `ModelExperimental` because models are only locally aware of their statistics, and `ResourceCache` is what now tracks the shared resources. We'll have to rewrite it for the `ModelExperimental` context.

File: Source/Scene/ModelExperimental/ModelExperimentalStatistics.js [hunk 3]
```diff
@@ -49,20 +50,51 @@ export default function ModelExperimentalStatistics() {
...
+
+  // Associated array of batch textures that have already been counted.
+  // This allows for quick look-up to check if a texture has been counted,
+  // while also allowing for dynamic texture counting.
+  this._batchTextureIdMap = new AssociativeArray();
 }
 
+Object.defineProperties(ModelExperimentalStatistics.prototype, {
+  /**
+   * Total size of the batch textures used for picking and styling.
```
Comment (ptrgags) [inline Source/Scene/ModelExperimental/ModelExperimentalStatistics.js:74]: I would add a sentence saying that this loops over batch textures to ensure the value is up-to-date (as batch textures are asynchronous)

File: Source/Scene/BatchTexture.js [hunk 4]
```diff
@@ -32,6 +33,8 @@ export default function BatchTexture(options) {
   Check.typeOf.object("options.owner", options.owner);
   //>>includeEnd('debug');
 
+  this._id = createGuid();
```
Comment (ptrgags) [inline Source/Scene/BatchTexture.js:36]: To answer your question, yes this is correct given that you're storing the batch textures in the statistics object :+1:

File: Source/Scene/ModelExperimental/ModelExperimentalStatistics.js (hunk 3 above)
Comment (ptrgags) [reply Source/Scene/ModelExperimental/ModelExperimentalStatistics.js:74]: Ah I see you did mention it in `addBatchTexture`, but wouldn't hurt to mention it here briefly

File: Specs/Scene/Cesium3DTilesetSpec.js [hunk 5]
```diff
@@ -1104,34 +1102,30 @@ describe(
...
 
     it("verify memory usage statistics for shared resources", function () {
+      ResourceCache.statistics.clear();
       // Six tiles total:
       // * Two b3dm tiles - no shared resources
       // * Two i3dm tiles with embedded glTF - no shared resources
       // * Two i3dm tiles with external glTF - shared resources
       // Expect to see some saving with memory usage since two of the tiles share resources
       // All tiles reference the same external texture but texture caching is not supported yet
       // TODO : tweak test when #5051 is in
```
Comment (ptrgags) [inline Specs/Scene/Cesium3DTilesetSpec.js:1112]: Given that #5051 was never merged and statistics work different, so maybe this line should be removed?

**Comment by j9liu** [comment]
@ptrgags updated!

**Comment by ptrgags** [comment]
The updates look good, thanks @j9liu!

**Pull request merged**